packages/ref-core/src/ref_core/datasets.py

```diff
@@ -8,3 +10,25 @@ class SourceDatasetType(enum.Enum):
 
     CMIP6 = "cmip6"
     CMIP7 = "cmip7"
+
+
+@frozen
+class FacetFilter:
+    """
+    A filter to apply to a data catalog of datasets.
+    """
+
+    facets: dict[str, str | tuple[str, ...]]
+    """
+    Filters to apply to the data catalog.
+
+    The keys are the metadata fields to filter on, and the values are the values to filter on.
+    If multiple values are provided for a single field, the filter will be applied as an OR operation.
+    Multiple filters are applied as an AND operation.
+    """
+    keep: bool = True
+    """
+    Whether to keep or remove datasets that match the filter.
+
+    If true (default), datasets that match the filter will be kept else they will be removed.
+    """
```

Comment:
Maybe it is worth to have a `IncludeFacetFilter` and another `ExcludeFacetFilter`. I sometimes tend to overlook the innocent `keep=False` after a long filter spec e.g. in scmdata.

Reply:
https://github.com/CMIP-REF/cmip-ref/issues/28